src/pytest_mysql/factories/client.py

```diff
@@ -131,8 +131,27 @@ def mysql_fixture(request: FixtureRequest) -> MySQLdb.Connection:
         yield mysql_conn
 
         # clean up after test that forgot to fetch selected data
-        mysql_conn.store_result()
-        mysql_conn.query("DROP DATABASE IF EXISTS %s" % mysql_db)
-        mysql_conn.close()
+        try:
+            if mysql_conn.open:
+                mysql_conn.store_result()
+        except Exception as e:
+            print(str(e))
+        try:
+            if not mysql_conn.open:
+                try:
+                    mysql_conn: MySQLdb.Connection = _connect(
+                        connection_kwargs, "DROP DATABASE IF EXISTS %s" % mysql_db
+                    )
+                except OperationalError:
+                    # Fallback to mysql connection with root user
+                    connection_kwargs["user"] = "root"
+                    mysql_conn: MySQLdb.Connection = _connect(
```

Comment:
I wonder why we would need a root user here... This would need investigation as it should not be required.

Reply:
Removed this